console/src/main/webapp/WEB-INF/views/changePasswordForm.jsp

```diff
@@ -41,91 +41,105 @@
 
 <body>
     <%@ include file="header.jsp" %>
+    <% String externalAuthentication=request.getHeader("sec-external-authentication");%>
```

Comment:
Don't we have access to the GeorchestraUser here, instead of having to work with the raw http headers ?

Reply:
We have access only to 'changePasswordForm'.
We need to handle access to this page globally (for OAuth2 and PreAuth cases). It would be better to use header to handle both cases, I think.

Reply:
grepping the codebase, I can't see anywhere else where we call `request.getHeader()` directly from a JSP file.

Taking what is done in `EditUserDetailsFormController` as an example, I'd have opted for putting the flag into the model ; also maybe introducing a constant into the SecurityHeaders class from the georchestra-commons would have been relevant, at least it would have been closer to how things are already done here.

* https://github.com/georchestra/georchestra/blob/master/console/src/main/java/org/georchestra/console/ws/edituserdetails/EditUserDetailsFormController.java#L115
* https://github.com/georchestra/georchestra/blob/master/commons/src/main/java/org/georchestra/commons/security/SecurityHeaders.java#L53-L65

Reply:
Hello,
  - request.getHeader() is not used in console but it has been used on header module
  - in EditUserDetailsFormController, we get user details from LDAP (EditUserDetailsFormController.java, line 116). We have already made the decision to not put external auth boolean into LDAP. So it is not possible to put this information into model object.
  - regarding SecurityHeaders constants, i have updated code in gateway and georchestra to use SecurityHeaders static constants.